src/runtime-rs/crates/kata/src/start.rs

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) 2021 Alibaba Cloud
+// Copyright (c) 2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::ffi::{OsStr, OsString};
+use std::io::Write;
+use std::os::unix::io::IntoRawFd;
+use std::os::unix::prelude::OsStrExt;
+use std::path::PathBuf;
+
+use unix_socket::UnixListener;
+
+use virtcontainers::spec_info;
+
+use crate::{Error, Result, ShimExecutor, KATA_BIND_FD};
+
+impl ShimExecutor {
+    // Implement start subcommnad
+    // This functions should be the most outside one for `start` command action
+    // So it does not return errors but directly write errors to stderr
+    pub fn start(&mut self) {
+        if let Err(e) = self.args.validate(false) {
+            eprintln!("start shim err: {} with invalid inputs {:?}", e, self.args);
+            return;
+        }
+
+        match self.do_start() {
+            Ok(address) => {
+                std::io::stdout()
+                    .write_all(address.as_os_str().as_bytes())
+                    .expect("failed to write stdout");
+            }
+            Err(e) => eprintln!("start shim err: {}", e),
```

Comment:
Same command as above: inconsistent error format so maybe creating a helper method that returns a `Result` would be clearer and make the code simpler here?